test/uk/gov/hmrc/tai/controllers/income/IncomeControllerSpec.scala

```diff
@@ -191,15 +200,452 @@ class IncomeControllerSpec extends PlaySpec
+      TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+        "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Live, BigDecimal(321.12), BigDecimal(0), BigDecimal(0)),
+      TaxCodeIncome(EmploymentIncome, Some(3), BigDecimal(0),
+        "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Live, BigDecimal(321.12), BigDecimal(0), BigDecimal(0)))
+
+    val employment = Employment("company name", Some("888"), new LocalDate(TaxYear().next.year, 5, 26),
+      None, Nil, "", "", 2, Some(100), hasPayrolledBenefit = false, receivingOccupationalPension = true)
+    val employments = Seq(employment, employment.copy(sequenceNumber = 1))
+
+    "return list of live and matched Employments & TaxCodeIncomes as IncomeSource for a given year" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Live")(FakeRequest())
+
+      status(result) mustBe OK
+
+      val expectedIncomeSource = Json.toJson(IncomeSource(TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+        "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Live, BigDecimal(321.12), BigDecimal(0), BigDecimal(0)),
+        employment
+      ))
+
+      val expectedJson = Json.obj(
+        "data" -> Json.arr(expectedIncomeSource),
+        "links" -> Json.arr()
+      )
+
+      contentAsJson(result) mustBe expectedJson
+    }
+
+    "return list of ceased and matched Employments & TaxCodeIncomes as IncomeSource for a given year" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(Seq(
+          TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+            "EmploymentIncome", "1100L", "Employer2", OtherBasisOperation, Ceased, BigDecimal(321.12), BigDecimal(0), BigDecimal(0))
+        )))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Ceased")(FakeRequest())
+
```

Comment:
Don't use strings for types here, "Ceased" is a real object used above and EmploymentIncome is a defined type elsewhere